pkg/azure/actuator.go

```diff
@@ -151,6 +168,17 @@ func (a *Actuator) needsUpdate(ctx context.Context, cr *minterv1.CredentialsRequ
 			logger.Debug("root cloud creds have changed, update is needed")
 			return true, nil
 		}
+	} else {
+		// mint-specific checks here
+
+		spItems, err := azureCredentialsMinter.spClient.List(ctx, fmt.Sprintf("appId eq '%v'", azureStatus.AppID))
+		if err != nil {
+			return true, err
+		}
+		if len(spItems) == 0 {
+			log.Errorf("ServicePrincipal %s does not exist, creating a new ServicePrincipal", azureStatus.ServicePrincipalName)
```

Comment:
I don't think I have a strong opinion here, but should this be an error or warning message?

Reply:
warning sounds better to me